openstates/de/legislators.py

```diff
@@ -16,27 +16,50 @@ def scrape(self, chamber, term, text=methodcaller('text_content'),
                re_spaces=re.compile(r'\s{,5}')):
 
         url = {
-            'upper': 'http://legis.delaware.gov/legislature.nsf/sen?openview&nav=senate',
-            'lower': 'http://legis.delaware.gov/legislature.nsf/Reps?openview&Count=75&nav=house&count=75',
+            'upper': 'http://legis.delaware.gov/legislature.nsf/sen?openview',
+            'lower': 'http://legis.delaware.gov/Legislature.nsf/Reps?openview',
             }[chamber]
 
-        doc = lxml.html.fromstring(self.urlopen(url))
+        doc = lxml.html.fromstring(self.get(url).text)
         doc.make_links_absolute(url)
 
-        # Sneak into the main table...
-        xpath = '//font[contains(., "Leadership Position")]/ancestor::table[1]'
-        table = doc.xpath(xpath)[0]
+        if chamber == "upper":
+            #for the senate, it's the same table
+            #but the html is hard-coded in js.
+            table_js = doc.xpath('.//script')[-1].text_content()
+            table = None
+            for line in table_js.split("\n"):
+                if line.strip().startswith("var") and "sen=" in line:
+                    table = line.replace("var","")
+                    table = table.replace('sen="<','<')
+                    table = table.replace('>";','>')
+                    break
 
-        # Skip the first tr (headings)
-        trs = table.xpath('tr')[1:]
+            assert table is not None, "Senate table could not be found"
+
+            table = lxml.html.fromstring(table)
+            table.make_links_absolute(url)
+            trs = table.xpath('//tr')
+
+
+        else:
+            #same table for the house, but kindly in actual html
+            trs = doc.xpath('//tr')
+
+        base_url = "http://legis.delaware.gov/"
 
         for tr in trs:
 
-            bio_url = tr.xpath('descendant::a/@href')[0]
-            name, _, district = map(text, tr.xpath("td"))
+            name_and_url = tr.xpath('.//a')[0]
+            bio_url = name_and_url.attrib["href"]
+            bio_url = bio_url.replace("JavaScript:window.location.href=","")
+            bio_url = base_url + bio_url.replace('"','')
```

Comment:
There's double-`/`s in this, since both `base_url[-1]` and `bio_url[0]` are `/`s.